modules/material/themes/material/silauth/loginuserpass.twig

```diff
@@ -0,0 +1,146 @@
+        var loginButton = document.querySelector('button');
+
+        grecaptcha.render(loginButton, {
+          sitekey: '{{ siteKey|e('js')|raw }}',
+          callback: submitForm
+        });
+      }
+
+      ga('send', 'event', 'reCAPTCHA', 'required');
+    </script>
+  {% endif %}
+</head>
+<body class="gradient-bg">
+<div class="mdl-layout mdl-layout--fixed-header fill-viewport">
+  <main class="mdl-layout__content" layout-children="column" child-spacing="center">
+    {% include 'announcement.twig' %}
+
+    <form method="post" autocomplete="off" onsubmit="event.target.onsubmit = preventDefault">
+      <input type="hidden" name="AuthState" value="{{ (stateparams.AuthState??'')|e }}"/>
+
+      {% if csrfToken is defined %}
+        <input type="hidden" name="csrf-token" value="{{ csrfToken|e }}"/>
+      {% endif %}
+
+      <div class="mdl-card mdl-shadow--8dp fill-phone-viewport">
+        <div class="mdl-card__media white-bg margin" layout-children="column">
+          <img src="/logo.png" alt="{{ '{login:logo}'|trans({'%idpName%': idpName}) }}">
+        </div>
+
+        <div class="mdl-card__title center">
+          <h1 class="mdl-card__title-text">
+            {{ '{login:header}'|trans({'%idpName%': idpName}) }}
+          </h1>
+        </div>
+
+        <div class="mdl-card__supporting-text" layout-children="column">
+          <div class="mdl-textfield mdl-js-textfield mdl-textfield--floating-label">
+            <label for="username" class="mdl-textfield__label">
+              {{ '{login:label_username}'|trans }}
+            </label>
+
+            <input
+              type="text"
+              required
+              name="username"
+              class="mdl-textfield__input mdl-color-text--black"
+              value="{{ username|e }}"
+              {% if username is empty %} autofocus {% endif %}
+              id="username"
+            >
+          </div>
+
+          <div class="mdl-textfield mdl-js-textfield mdl-textfield--floating-label">
+            <label for="password" class="mdl-textfield__label">
+              {{ '{login:label_password}'|trans }}
+            </label>
+
+            <input
+              type="password"
+              required
+              name="password"
+              class="mdl-textfield__input mdl-color-text--black"
+              {% if not username is empty %} autofocus {% endif %}
+              id="password"
+            >
+          </div>
+        </div>
+
+        {% if errorCode ?? '' == 'WRONGUSERPASS' %}
+          {% set errorMessageKey = errorparams[1] ?? '{login:error_wronguserpass}'|trans %}
+          {% set errorMessageTokens = errorparams[2] ?? null %}
+          {% set message = errorMessageKey|trans({errorMessageKey, errorMessageTokens}) %}
+          <p class="mdl-card__supporting-text mdl-color-text--red error">
+            <i class="material-icons">error</i>
+
+            <span class="mdl-textfield mdl-typography--caption">
+              {{ message|e }}
```

Comment:
Optional. I think you could simplify it, but if you think it's clearer as is, feel free to leave it.

```suggestion
          <p class="mdl-card__supporting-text mdl-color-text--red error">
            <i class="material-icons">error</i>

            <span class="mdl-textfield mdl-typography--caption">
              {{ errorMessageKey|trans({errorMessageKey, errorparams[2] ?? null}) }}
```

Reply:
Thanks, but I think it's clear as is.